packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -622,11 +551,16 @@ add (const Scalar& alpha,
 {
   typedef Map<LocalOrdinal,GlobalOrdinal,Node>                     map_type;
   typedef CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>  crs_matrix_type;
+  //Check that A,B are fillComplete before getting B's column map
+  TEUCHOS_TEST_FOR_EXCEPTION
+    (! A.isFillComplete () || ! B.isFillComplete (), std::invalid_argument,
+     "TpetraExt::MatrixMatrix::add(): A and B must both be fill complete.");
   Teuchos::RCP<const map_type> CrowMap = transposeB ?  B.getDomainMap() : B.getRowMap();
-
   Teuchos::RCP<crs_matrix_type> C = rcp(new crs_matrix_type(CrowMap, 0));
-
-  add(alpha,transposeA,A,beta,transposeB,B,*C,domainMap,rangeMap,params);
+  //this version of add() always fill completes the result
+  if(!params.is_null())
```

Comment:
You shouldn't need to say `<bool>` here.  The following should suffice, since it can deduce the template argument from the method's second argument here.
```c++
params->set("Call fillComplete", true);
```